server/grpc_framework_server.go

```diff
@@ -139,7 +141,7 @@ func (s *GrpcFrameworkServer) Start() error {
 	// use caller's authN interceptor if provided
 	if s.config.AuthNUnaryInterceptor != nil {
 		unaryInterceptors = append(unaryInterceptors, s.config.AuthNUnaryInterceptor)
-	} else if len(s.config.Security.Authenticators) > 0 {
+	} else if s.config.Security.Authenticators != nil && len(s.config.Security.Authenticators.ListIssuersWithClientID()) > 0 {
```

Comment:
nit: We could set the `s.config.Security.Authenticators` to nil if there are no issuers. Then, we don't have to check for both conditions everywhere.